component/componentstatus/instance.go

```diff
@@ -66,9 +81,44 @@ func (id *InstanceID) AllPipelineIDs(f func(component.ID) bool) {
 	}
 }
 
+// AllPipelineIDsWithPipelineIDs calls f for each pipeline this instance is associated with. If
```

Comment:
I can't think of a better name but this made me laugh :smile:

Reply:
Yes, the connector interfaces are the same.